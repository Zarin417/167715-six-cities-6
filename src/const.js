export const CITIES = [
  `Paris`,
  `Cologne`,
  `Brussels`,
  `Amsterdam`,
  `Hamburg`,
  `Dusseldorf`
];

export const CardName = {
  CITIES: `cities`,
  FAVORITES: `favorites`,
  NEAR_PLACES: `near-places`,
};

export const CardsListName = {
  NEAR_PLACES_LIST: `near-places__list`,
  CITIES_PLACES_LIST: `cities__places-list tabs__content`
};

export const ButtonName = {
  PROPERTY: `property`,
  PLACE_CARD: `place-card`,
  FAVORITE: `favorite`,
  NEAR_PLACE: `near-place`
};

export const SortType = {
  POPULAR: `popular`,
  PRICE_LOW_TO_HIGH: `price: low to high`,
  PRICE_HIGH_TO_LOW: `price: high to low`,
  TOP_RATED: `top rated first`
};

export const AppRoute = {
  ROOT: `/`,
  LOGIN: `/login`,
  FAVORITES: `/favorites`,
  CITY: `/city/:city`,
  OFFER: `/offer/:id`,
  ERROR: `/error`
};

export const APIRoute = {
  HOTELS: `/hotels`,
  LOGIN: `/login`,
  LOGOUT: `/logout`,
  COMMENTS: `/comments`,
  FAVORITE: `/favorite`,
};
