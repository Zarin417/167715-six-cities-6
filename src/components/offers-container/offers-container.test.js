import React from "react";
import {render} from "@testing-library/react";
import configureStore from "redux-mock-store";
import {createMemoryHistory} from "history";
import {Provider} from "react-redux";
import OffersContainer from "./offers-container";
import {SortType} from "../../const";
import {Router} from "react-router";

const mockStore = configureStore({});

describe(`Test 'OffersContainer'`, () => {
  it(`OffersContainer should render correctly`, () => {
    const history = createMemoryHistory();
    const store = mockStore({
      SORT: {sortType: SortType.PRICE_HIGH_TO_LOW},
      CARD: {activeCard: 15},
      PLACE_INFO: {placeInfo: {}},
      USER: {isLoggedIn: true},
      OFFER: {
        activeCity: `Paris`,
        offers: [
          {
            title: `The house among olive`,
            price: 169,
            previewImage: ``,
            type: `hotel`,
            isFavorite: true,
            isPremium: true,
            rating: 5,
            id: 15,
            bedrooms: 2,
            description: ``,
            maxAdults: 2,
            location: {latitude: 48.83861, longitude: 2.350499, zoom: 16},
            images: [``],
            host: {id: 25, name: `Maria`, isPro: true, avatarUrl: ``},
            goods: [`Breakfast`],
            city: {name: `Paris`, location: {latitude: 48.85661, longitude: 2.351499, zoom: 13}},
          },
          {
            title: `The house among olive`,
            price: 169,
            previewImage: ``,
            type: `hostel`,
            isFavorite: false,
            isPremium: false,
            rating: 5,
            id: 19,
            bedrooms: 2,
            description: ``,
            maxAdults: 2,
            location: {latitude: 48.837610000000005, longitude: 2.3454990000000002, zoom: 16},
            images: [``],
            host: {id: 25, name: `Helga`, isPro: true, avatarUrl: ``},
            goods: [`Wi-Fi`],
            city: {name: `Paris`, location: {latitude: 48.85661, longitude: 2.351499, zoom: 13}},
          }
        ]
      }
    });

    const {container} = render(
        <Provider store={store}>
          <Router history={history}>
            <OffersContainer />
          </Router>
        </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
