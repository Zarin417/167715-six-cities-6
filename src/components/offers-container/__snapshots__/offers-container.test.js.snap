// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test 'OffersContainer' OffersContainer should render correctly 1`] = `
<div>
  <div
    class="cities__places-container container"
  >
    <section
      class="cities__places places"
    >
      <h2
        class="visually-hidden"
      >
        Places
      </h2>
      <b
        class="places__found"
      >
        2
              places
            to stay in Paris
      </b>
      <form
        action="#"
        class="places__sorting"
        method="get"
      >
        <span
          class="places__sorting-caption"
        >
          Sort by 
        </span>
        <span
          class="places__sorting-type"
          tabindex="0"
        >
          Price: high to low
          <svg
            class="places__sorting-arrow"
            height="4"
            width="7"
          >
            <use
              xlink:href="#icon-arrow-select"
            />
          </svg>
        </span>
        <ul
          class="places__options places__options--custom false"
        >
          <li
            class="places__option false"
            tabindex="0"
            type="popular"
          >
            Popular
          </li>
          <li
            class="places__option false"
            tabindex="0"
            type="price: low to high"
          >
            Price: low to high
          </li>
          <li
            class="places__option places__option--active"
            tabindex="0"
            type="price: high to low"
          >
            Price: high to low
          </li>
          <li
            class="places__option false"
            tabindex="0"
            type="top rated first"
          >
            Top rated first
          </li>
        </ul>
      </form>
      <div
        class="cities__places-list tabs__content places__list"
      >
        <article
          class="cities__place-card place-card"
        >
          <div
            class="place-card__mark"
          >
            <span>
              Premium
            </span>
          </div>
          <div
            class="cities__image-wrapper place-card__image-wrapper"
          >
            <a
              href="/offer/15"
            >
              <img
                alt="Place image"
                class="place-card__image"
                height="200"
                src=""
                width="260"
              />
            </a>
          </div>
          <div
            class=" place-card__info"
          >
            <div
              class="place-card__price-wrapper"
            >
              <div
                class="place-card__price"
              >
                <b
                  class="place-card__price-value"
                >
                  €
                  169
                </b>
                <span
                  class="place-card__price-text"
                >
                  / night
                </span>
              </div>
              <button
                class="place-card__bookmark-button
      place-card__bookmark-button--active button"
                type="button"
              >
                <svg
                  class="place-card__bookmark-icon"
                  height="19"
                  width="18"
                >
                  <use
                    xlink:href="#icon-bookmark"
                  />
                </svg>
                <span
                  class="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              class="place-card__rating rating"
            >
              <div
                class="place-card__stars rating__stars"
              >
                <span
                  style="width: 100%;"
                />
                <span
                  class="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              class="place-card__name"
            >
              <a
                href="/offer/15"
              >
                The house among olive
              </a>
            </h2>
            <p
              class="place-card__type"
            >
              Hotel
            </p>
          </div>
        </article>
        <article
          class="cities__place-card place-card"
        >
          <div
            class="cities__image-wrapper place-card__image-wrapper"
          >
            <a
              href="/offer/19"
            >
              <img
                alt="Place image"
                class="place-card__image"
                height="200"
                src=""
                width="260"
              />
            </a>
          </div>
          <div
            class=" place-card__info"
          >
            <div
              class="place-card__price-wrapper"
            >
              <div
                class="place-card__price"
              >
                <b
                  class="place-card__price-value"
                >
                  €
                  169
                </b>
                <span
                  class="place-card__price-text"
                >
                  / night
                </span>
              </div>
              <button
                class="place-card__bookmark-button
       button"
                type="button"
              >
                <svg
                  class="place-card__bookmark-icon"
                  height="19"
                  width="18"
                >
                  <use
                    xlink:href="#icon-bookmark"
                  />
                </svg>
                <span
                  class="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              class="place-card__rating rating"
            >
              <div
                class="place-card__stars rating__stars"
              >
                <span
                  style="width: 100%;"
                />
                <span
                  class="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              class="place-card__name"
            >
              <a
                href="/offer/19"
              >
                The house among olive
              </a>
            </h2>
            <p
              class="place-card__type"
            >
              Hostel
            </p>
          </div>
        </article>
      </div>
    </section>
    <div
      class="cities__right-section"
    >
      <section
        class="cities__map map"
      >
        <div
          class="leaflet-container leaflet-grab leaflet-touch-drag"
          id="map"
          style="height: 100%; position: relative;"
          tabindex="0"
        >
          <div
            class="leaflet-pane leaflet-map-pane"
            style="left: 0px; top: 0px;"
          >
            <div
              class="leaflet-pane leaflet-tile-pane"
            >
              <div
                class="leaflet-layer "
                style="z-index: 1;"
              >
                <div
                  class="leaflet-tile-container leaflet-zoom-animated"
                  style="z-index: 18; left: 0px; top: 0px;"
                >
                  <img
                    alt=""
                    class="leaflet-tile"
                    role="presentation"
                    src="https://b.basemaps.cartocdn.com/rastertiles/voyager/13/4149/2818.png"
                    style="width: 256px; height: 256px; left: -130px; top: -72px;"
                  />
                </div>
              </div>
            </div>
            <div
              class="leaflet-pane leaflet-shadow-pane"
            />
            <div
              class="leaflet-pane leaflet-overlay-pane"
            />
            <div
              class="leaflet-pane leaflet-marker-pane"
            >
              <img
                alt=""
                class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
                src="img/pin-active.svg"
                style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: -5px; top: 159px; z-index: 159;"
                tabindex="0"
              />
              <img
                alt=""
                class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
                src="img/pin.svg"
                style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: -34px; top: 168px; z-index: 168;"
                tabindex="0"
              />
            </div>
            <div
              class="leaflet-pane leaflet-tooltip-pane"
            />
            <div
              class="leaflet-pane leaflet-popup-pane"
            />
          </div>
          <div
            class="leaflet-control-container"
          >
            <div
              class="leaflet-top leaflet-left"
            />
            <div
              class="leaflet-top leaflet-right"
            />
            <div
              class="leaflet-bottom leaflet-left"
            />
            <div
              class="leaflet-bottom leaflet-right"
            >
              <div
                class="leaflet-control-attribution leaflet-control"
              >
                <a
                  href="https://leafletjs.com"
                  title="A JS library for interactive maps"
                >
                  Leaflet
                </a>
                 | © 
                <a
                  href="https://www.openstreetmap.org/copyright"
                >
                  OpenStreetMap
                </a>
                
          contributors © 
                <a
                  href="https://carto.com/attributions"
                >
                  CARTO
                </a>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </div>
</div>
`;
